### PR TITLE
chore(tests/utils): suppress error boundary error log

### PR DESCRIPTION
**Pull request by tmkx**
## Summary

it's a bit verbose when debugging :)

image

## Check List

- [x] `yarn run prettier` for formatting code and docs

### CODE REVIEW

File: tests/react/vanilla-utils/atomWithObservable.test.tsx [hunk 1]
```diff
@@ -17,11 +26,14 @@ beforeEach(() => {
     performance.clearMarks = (() => {}) as any
     performance.clearMeasures = (() => {}) as any
   }
+  // suppress ErrorBoundary error log
+  consoleErrorSpy = vi.spyOn(console, 'error').mockImplementation(() => {})
```
Comment (dai-shi) [inline tests/react/vanilla-utils/atomWithObservable.test.tsx:30]: Nice idea. Can we only suppress the error boundary error, using some simple regex?

We have this, so maybe we should be consistent about using `spyOn`?
https://github.com/pmndrs/jotai/blob/ccba918281069f3ce003988f591389df4fefdde5/tests/react/error.test.tsx#L12

Comment (tmkx) [reply tests/react/vanilla-utils/atomWithObservable.test.tsx]: atomWithObservable was excluded from older ts test, so it passed, but `error.test.tsx` it not:

https://github.com/pmndrs/jotai/blob/ccba918281069f3ce003988f591389df4fefdde5/.github/workflows/test-old-typescript.yml#L69

Comment (dai-shi) [reply tests/react/vanilla-utils/atomWithObservable.test.tsx]: Hmmm, maybe we give up `spyOn` and just assign `console.error = ...`?

**Review by dai-shi — APPROVED** [review]
LGTM!

**Pull request merged**